transports/rabbitmq/delayed-delivery.md

```diff
@@ -21,58 +21,111 @@ N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.N.destination
 
-exchangeN(Level N)
-queueN[TTL 2^N]
-exchangeN-1(Level N-1)
+exchange27(Exchange: nsb.delay-level-27)
+queue27[Queue: nsb.delay-level-27]
+exchange26(Exchange: nsb.delay-level-26)
+queue26[Queue: nsb.delay-level-26]
+exchange25(Exchange: nsb.delay-level-25)
 
-exchangeN -->|bit=1| queueN
-exchangeN -->|bit=0| exchangeN-1
-queueN -->|expired<br>message| exchangeN-1
+hiddenExchanges(Not Shown: nsb.delay-level-24 through nsb.delay-level-2)
 
-classDef exchangeClass stroke:#000000,stroke-width:2px;
-class exchangeN,exchangeN-1 exchangeClass
+exchange1(Exchange: nsb.delay-level-01)
+queue1(Queue: nsb.delay-level-01)
+exchange0(Exchange: nsb.delay-level-00)
+queue0(Queue: nsb.delay-level-00)
+delivery(Exchange: nsb.delay-delivery)
+
+exchange27 -->|"Binding: 1.#<br/>(No * wildcards)"| queue27
+exchange27 -->|"Binding: 0.#<br/>(No * wildcards)"| exchange26
+queue27 -->|"TTL: 2^27s ≈ 4.25 years"| exchange26
+
+exchange26 -->|"Binding: *.1.#<br/>(1 * wildcard)"| queue26
+exchange26 -->|"Binding: *.0.#<br/>(1 * wildcard)"| exchange25
+queue26 -->|"TTL: 2^26s ≈ 2.13 years"| exchange25
+
+exchange25 -.-> hiddenExchanges
+hiddenExchanges -.-> exchange1
+
+exchange1 -->|"Binding:<br/>*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.1.#<br/>(26 * wildcards)"| queue1
+exchange1 -->|"Binding:<br/>*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.0.#<br/>(26 * wildcards)"| exchange0
+queue1 -->|TTL: 2^1s = 2 seconds| exchange0
+
+exchange0 -->|"Binding:<br/>*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.1.#<br/>(27 * wildcards)"| queue0
+exchange0 -->|"Binding:<br/>*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.*.0.#<br/>(27 * wildcards)"| delivery
+queue0 -->|TTL: 2^0s = 1 second| delivery
+
+classDef exchangeClass stroke:#000000,stroke-width:2px,font-weight:bold;
+class exchange27,exchange26,exchange25,exchange1,exchange0,delivery exchangeClass
+
+classDef hiddenExchanges font-style:italic,fill:transparent,stroke:transparent;
+class hiddenExchanges hiddenExchanges
 
 end
 ```
 
-The delay levels are connected in this manner, from highest (27) to lowest (0). Each delay level's routing key's add wildcards as needed so that they are looking at the portion of the message's routing key that corresponds to its delay level.
+With this system of exchanges and queues, the message is first delivered to the first applicable exchange based on the binary value. In the 10-second example (binary `1010`) the message would first be delivered to the `nsb.delay-level-03` exchange, and all of that exchange's bindings are evaluated. All matching bindings will receive a copy of the message, but the bindings are arranged so that there are only two possibilities based on the value of the 4th-to-last digit:
+
+* If the value is a `1` the exchange will deliver the message to the `nsb.delay-level-03` queue and wait for something to process it up until the TTL (2^3 seconds) before forwarding to the `nsb-delay-level-02` exchange.
```

Comment:
Needs comma after "`1`"

Reply:
deliver -> route

Reply:
"wait for something to process it" We don't actually expect this to happen, so I'd drop this and just say

"will deliver the message to the `nsb.delay-level-03` queue, where it will wait until the TTL expires (2^3 seconds) before being forwarded..."

Reply:
Putting all the suggestions together:

If the value is a `1`, the exchange will route the message to the `nsb.delay-level-03` queue, where it will wait until the TTL expires (2^3 seconds) before being forwarded to the `nsb-delay-level-02` exchange.